Mathlib/Algebra/Lie/Killing.lean

```diff
@@ -60,13 +64,25 @@ lemma killingForm_nondegenerate :
     (killingForm R L).Nondegenerate := by
   simp [LinearMap.BilinForm.nondegenerate_iff_ker_eq_bot]
 
-/-- The converse of this is true over a field of characteristic zero. There are counterexamples
-over fields with positive characteristic. -/
-instance instHasTrivialRadical [IsDomain R] [IsPrincipalIdealRing R] : HasTrivialRadical R L := by
-  refine' (hasTrivialRadical_iff_no_abelian_ideals R L).mpr fun I hI ↦ _
+variable {R L} in
+lemma ideal_eq_bot_of_isLieAbelian [IsDomain R] [IsPrincipalIdealRing R]
+    (I : LieIdeal R L) [IsLieAbelian I] : I = ⊥ := by
   rw [eq_bot_iff, ← killingCompl_top_eq_bot]
   exact I.le_killingCompl_top_of_isLieAbelian
 
+instance instSemisimple [IsKilling K L] [Module.Finite K L] : IsSemisimple K L := by
+  apply InvariantForm.isSemisimple_of_nondegenerate (Φ := killingForm K L)
+  · exact IsKilling.killingForm_nondegenerate _ _
+  · exact LieModule.traceForm_apply_lie_apply K L L
+  · exact (LieModule.traceForm_isSymm K L L).isRefl
+  · intro I h₁ h₂
+    exact h₁.1 <| IsKilling.ideal_eq_bot_of_isLieAbelian I
+
+/-- The converse of this is true over a field of characteristic zero. There are counterexamples
+over fields with positive characteristic. -/
+instance instHasTrivialRadical [IsDomain R] [IsPrincipalIdealRing R] : HasTrivialRadical R L :=
+  (hasTrivialRadical_iff_no_abelian_ideals R L).mpr IsKilling.ideal_eq_bot_of_isLieAbelian
+
```

Comment:
I think I'd drop this instance now (even though it's technically slightly stronger than our results over fields)
```suggestion
```

Reply:
I actually think we should keep it! Even though we don't have a use in the near future, you never know...